PWGGA/GammaConv/AliAnalysisTaskGammaCalo.cxx

```diff
@@ -6084,8 +6086,21 @@ void AliAnalysisTaskGammaCalo::ProcessTrueMesonCandidates(AliAODConversionMother
   }
   if (TrueGammaCandidate1->GetIsCaloPhoton() == 0) AliFatal("CaloPhotonFlag has not been set. Aborting");
 
+  Bool_t previouslyNotFoundTrueMesons = kFALSE;
```

Comment:
probably will not/rarely happen but maybe a good idea to put a depth counter to avoid infinite loops. If I recall correctly it can sometimes happen for some MCs that there are loops in the mother daughter chain and then this while loop goes on forever so maybe depth = 20 and increase a counter during each while loop

Reply:
ok I can do that